java/src/com/ibm/streamsx/topology/internal/context/ZippedToolkitStreamsContext.java

```diff
@@ -0,0 +1,84 @@
+package com.ibm.streamsx.topology.internal.context;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import com.ibm.json.java.JSONObject;
+import com.ibm.streamsx.topology.Topology;
+import com.ibm.streamsx.topology.context.ContextProperties;
+import com.ibm.streamsx.topology.internal.process.CompletedFuture;
+
+public class ZippedToolkitStreamsContext extends ToolkitStreamsContext {
+	
+	@Override
+	public Future<File> submit(Topology app, Map<String, Object> config) throws Exception {
+
+        if (config == null)
```

Comment:
Why are lines 28 through 3^ needed, isn't it exactly what the `super.submit()` call does? Why the need to repeat them?